contracts/SushiYieldSource.sol

```diff
@@ -1,94 +1,135 @@
+        address _sushiBarAddress = address(sushiBar);
+        uint256 allowance = sushiAddr.allowance(address(this), _sushiBarAddress);
+
+        sushiAddr.safeIncreaseAllowance(_sushiBarAddress, type(uint256).max.sub(allowance));
+        return true;
     }
 
     /// @notice Returns the ERC20 asset token used for deposits.
     /// @return The ERC20 asset token
-    function depositToken() public view override returns (address) {
+    function depositToken() external view override returns (address) {
         return address(sushiAddr);
     }
 
     /// @notice Returns the total balance (in asset tokens).  This includes the deposits and interest.
     /// @return The underlying balance of asset tokens
-    function balanceOfToken(address addr) public override returns (uint256) {
+    function balanceOfToken(address addr) external override returns (uint256) {
         if (balances[addr] == 0) return 0;
 
         uint256 totalShares = sushiBar.totalSupply();
         uint256 barSushiBalance = sushiAddr.balanceOf(address(sushiBar));
 
-        return balances[addr].mul(barSushiBalance).div(totalShares);       
+        return balances[addr].mul(barSushiBalance).div(totalShares);
     }
 
     /// @notice Allows assets to be supplied on other user's behalf using the `to` param.
     /// @param amount The amount of `token()` to be supplied
     /// @param to The user whose balance will receive the tokens
-    function supplyTokenTo(uint256 amount, address to) public override {
-        sushiAddr.transferFrom(msg.sender, address(this), amount);
-        sushiAddr.approve(address(sushiBar), amount);
+    function supplyTokenTo(uint256 amount, address to) external override {
+        sushiAddr.safeTransferFrom(msg.sender, address(this), amount);
 
         ISushiBar bar = sushiBar;
         uint256 beforeBalance = bar.balanceOf(address(this));
-        
+
         bar.enter(amount);
-        
+
         uint256 afterBalance = bar.balanceOf(address(this));
         uint256 balanceDiff = afterBalance.sub(beforeBalance);
-        
+
         balances[to] = balances[to].add(balanceDiff);
+        emit SuppliedTokenTo(msg.sender, balanceDiff, amount, to);
     }
 
     /// @notice Redeems tokens from the yield source to the msg.sender, it burns yield bearing tokens and returns token to the sender.
     /// @param amount The amount of `token()` to withdraw.  Denominated in `token()` as above.
     /// @dev The maxiumum that can be called for token() is calculated by balanceOfToken() above.
-    /// @return The actual amount of tokens that were redeemed. This may be different from the amount passed due to the fractional math involved. 
-    function redeemToken(uint256 amount) public override returns (uint256) {
+    /// @return The actual amount of tokens that were redeemed. This may be different from the amount passed due to the fractional math involved.
+    function redeemToken(uint256 amount) external override returns (uint256) {
```

Comment:
could use a reentrancy guard....is this covered elsewhere?

Reply:
In this PR yes: https://github.com/pooltogether/sushi-pooltogether/pull/10